src/fsharp/fsi/console.fs

```diff
@@ -17,7 +17,7 @@ module internal ConsoleOptions =
   let fixupRequired = false
 #else
   // Bug 4254 was fixed in Dev11 (Net4.5), so this flag tracks making this fix up version specific.
-  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove
+  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && not runningOnMono
```

Comment:
What about using `System.Console.InputEncoding.IsSingleByte` here? (I.e., `let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && System.Console.InputEncoding.IsSingleByte`). Because I could see this failing on .Net Core, depending on what version it reports for the `System` DLL. https://docs.microsoft.com/en-us/dotnet/articles/core/versions/ appears to promise that the `System` DLL will report a version higher than 4.5, so in theory there's no danger of `IsRunningOnNetFx45OrAbove` becoming false on .Net Core. But I haven't yet been able to test that theory.

And using `System.Console.InputEncoding.IsSingleByte` would specifically target the buggy environments where the `readKeyFixup` function was actually necessary: console encodings set to single-byte (i.e., non-Unicode) code pages. That way, someone on Windows 7 with .Net 4.0, but who ran `chcp 65001` before running `fsi.exe`, would get the benefit of this bugfix, rather than having his UTF-8 console fail him. (With just the `not runningOnMono` check, our theoretical `chcp 65001` user would still suffer this bug).